File: feedback-service/src/services/rabbitmq.service.js
// Purpose: Provide reusable service/business logic.

const { getChannel } = require('../config/rabbitmq');

// Main flow: Execute core operations and return results.

// Function: publishEvent - Publishes event.
async function publishEvent(exchangeName, payload) {
  const channel = getChannel();
  await channel.assertExchange(exchangeName, 'fanout', { durable: true });
  channel.publish(exchangeName, '', Buffer.from(JSON.stringify(payload)));
}

module.exports = { publishEvent };

File: transcription-service/app/services/rabbitmq_service.py
# Purpose: Provide reusable service/business logic.

import os
import json
import pika


# Main flow: Execute core operations and return results.

# Function: publish_transcript_ready - Publishes transcript ready.
def publish_transcript_ready(payload: dict):
    url = os.getenv('RABBITMQ_URL', 'amqp://localhost')
    connection = pika.BlockingConnection(pika.URLParameters(url))
    channel = connection.channel()

    exchange = 'transcript.ready'
    channel.exchange_declare(exchange=exchange, exchange_type='fanout', durable=True)
    channel.basic_publish(
        exchange=exchange,
        routing_key='',
        body=json.dumps(payload),
    )
    connection.close()
